### DISCUSSION TITLE
La méthode select de PDOPlusPlus retourne un integer au lieu d'un array

### DISCUSSION THREAD

**Discussion by laurentsc** [rawsrc/PDOPlusPlus#5] [Q&A] [answered]
Même si tout est en anglais, j'écris en français !
Il s'agit probablement d'une mauvaise utilisation de ma part de PDOPlusPlus, mais le comportement que j'ai est que la méthode select ne retourne qu'un integer alors qu'elle devrait retourner un array de 2 integers...
Le code qui pose problème est : ` public function getAppliKeyFromSESA($sesa)
    {
        $ppp = new PDOPlusPlus( );
        $sql="SELECT application_key from `license` where sesa={$ppp($sesa)}";
        $res = $ppp->select($sql)[0]['application_key'];
        return $res;
    }`
Ici, la requête SQL retourne un array. En debug pas à pas, je constate que le FetchAll de la méthode select retourne bien un array, mais quand je récupère le résultat dans la variable $res,  ce n'est qu'un integer. Puis-je savoir où se situe le problème ?

**Comment by rawsrc** [comment, marked as answer]
Salut,

si tu veux récupérer le résultat de `->fetchAll()`, il suffit de faire `$res = $ppp->select($sql);`

**Reply by laurentsc** [reply]
merci
